app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1289,23 +1289,25 @@ public void onClick(View v) throws NoSuchElementException {
                 });
             }
 
-            cardOptions.put(getString(R.string.takePhoto), () -> {
-                int permissionRequestType;
-
-                if (v.getId() == R.id.frontImageHolder) {
-                    permissionRequestType = PERMISSION_REQUEST_CAMERA_IMAGE_FRONT;
-                } else if (v.getId() == R.id.backImageHolder) {
-                    permissionRequestType = PERMISSION_REQUEST_CAMERA_IMAGE_BACK;
-                } else if (v.getId() == R.id.thumbnail) {
-                    permissionRequestType = PERMISSION_REQUEST_CAMERA_IMAGE_ICON;
-                } else {
-                    throw new IllegalArgumentException("Unknown ID type " + v.getId());
-                }
+            if (Utils.deviceHasCamera(LoyaltyCardEditActivity.this)) {
+                cardOptions.put(getString(R.string.takePhoto), () -> {
+                    int permissionRequestType;
+
+                    if (v.getId() == R.id.frontImageHolder) {
+                        permissionRequestType = PERMISSION_REQUEST_CAMERA_IMAGE_FRONT;
+                    } else if (v.getId() == R.id.backImageHolder) {
+                        permissionRequestType = PERMISSION_REQUEST_CAMERA_IMAGE_BACK;
+                    } else if (v.getId() == R.id.thumbnail) {
+                        permissionRequestType = PERMISSION_REQUEST_CAMERA_IMAGE_ICON;
+                    } else {
+                        throw new IllegalArgumentException("Unknown ID type " + v.getId());
+                    }
 
-                PermissionUtils.requestCameraPermission(LoyaltyCardEditActivity.this, permissionRequestType);
+                    PermissionUtils.requestCameraPermission(LoyaltyCardEditActivity.this, permissionRequestType);
 
-                return null;
-            });
+                    return null;
+                });
+            }
```

Comment:
I'm thinking this "has camera" check should probably be removed as it opens the camera app and will either show a toast because there isn't one or users will have other apps installed that can fake the camera and allow them to do useful stuff